modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/TableRowSkinTest.java

```diff
@@ -244,6 +247,47 @@ public void tableRowWithFixedSizeShouldIgnoreVerticalPadding() {
         assertEquals(height, cell.getHeight(), 0);
     }
 
+    /**
+     * When we make an invisible column visible we expect the underlying cells to be visible, e.g. as width > 0.
+     * See also: JDK-8305248
+     */
+    @Test
+    public void testMakeInvisibleColumnVisible() {
+        tableView.setFixedCellSize(24);
+        TableColumn<Person, ?> firstColumn = tableView.getColumns().get(0);
+        firstColumn.setVisible(false);
+
+        tableView.refresh();
+        Toolkit.getToolkit().firePulse();
+
+        firstColumn.setVisible(true);
+        Toolkit.getToolkit().firePulse();
+
+        IndexedCell<?> row = VirtualFlowTestUtils.getCell(tableView, 0);
+        for (Node cellNode : row.getChildrenUnmodifiable()) {
+            double width = ((Region) cellNode).getWidth();
+            assertNotEquals(0.0, width);
+        }
+    }
+
+    @Test
+    public void testMakeVisibleColumnInvisible() {
+        tableView.setFixedCellSize(24);
+        TableColumn<Person, ?> firstColumn = tableView.getColumns().get(0);
```

Comment:
Might be unnecessary, but you could assert that the first column was visible, just to make sure that the next call to make it invisible has a real change?